### ISSUE TITLE
Classification Error

### ISSUE THREAD

**Issue by sainisanjay** [facebook/C3D#273]
```
F0529 14:49:04.880832 18092 net.cpp:757] Cannot copy param 0 weights from layer 'fc8-new'; shape mismatch.  **Source param shape is 25 512 (12800); target param shape is 25 5120 (128000).** To learn this layer's parameters from scratch rather than copying from a saved net, rename the layer.
*** Check failure stack trace: ***
    @     0x7f9f04cfedaa  (unknown)
    @     0x7f9f04cfece4  (unknown)
    @     0x7f9f04cfe6e6  (unknown)
    @     0x7f9f04d01687  (unknown)
    @     0x7f9f07f23407  caffe::Net<>::CopyTrainedLayersFrom()
    @     0x7f9f07f2ca32  caffe::Net<>::CopyTrainedLayersFromBinaryProto()
    @     0x7f9f07f2ca96  caffe::Net<>::CopyTrainedLayersFrom()
    @           0x4075f5  Classifier::Classifier()
    @           0x403ffb  main
    @     0x7f9f04217f45  (unknown)
    @           0x40450b  (unknown)
    @              (nil)  (unknown)
./eval_c3d_nvidia2.sh: line 1: 18092 Aborted                 (core dumped) ../../build/examples/cpp_classification/classification.bin deploy_resnet18.prototxt c3d_resnet18_ucf101_r2_ft_iter_100000.caffemodel 128,128,128 test_list.txt label_list.txt
```
I can see from error that source and target param shape not matching but I have cross checked in both train and deploy.prototxt file, i am using same input data value. As can see below.

**deploy.prototxt file** 
name: "C3D"
input: "data"
layer {
  name: "data"
  type: "Input"
  top: "data"
  input_param { shape: { dim: 1 dim: 3 dim: 8 dim: 128 dim: 171 } }
}
'.......
//All remaining layers are same// 
.......

# ----- 1st fc group -----
layer {
  name: "fc8-new"
  type: "InnerProduct"
  bottom: "pool5"
  top: "fc8-new"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  inner_product_param {
    num_output: 25
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}

layer {
  name: "prob"
  type: "Softmax"
  bottom: "fc8-new"
  top: "prob"
}

**Train prototxt:** 
name: "C3D"
layer {
  name: "data"
  type: "VideoData"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  transform_param {
    crop_size: 112
    mean_value: 128
    mirror: true
  }
  video_data_param {
    source: "nvidia_train_list.txt"
    new_length: 8
    sampling_rate: 2
    new_height: 128
    new_width: 171
    use_image: true
    show_data: false
    batch_size: 8
    shuffle: true
  }
}
........

**Comment by dutran** [comment]
This tool (cpp_classification/classification.bin) is still assume 2D convnet. It needs to be modified to adapt for 3D convnet.

**Comment by sainisanjay** [comment]
Thanks @dutran I realized the same, and thanks a lot for your response.
